transducers/shared/src/main/scala/fs2/data/stt/Assignment.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.data.stt
+
+import cats.Show
+import cats.syntax.show._
+
+sealed trait Assignment[+V <: Variable, +C]
+sealed trait Reset[+C] extends Assignment[Nothing, C]
+object Reset {
+  def unapply[C](r: Reset[C]): Some[Reset[C]] =
```

Comment:
Is this really needed? Shouldn't it be a irrefutable pattern anyways and get handled by the compiler?

Reply:
How do you plan on writing `case Reset(r) =>` if there is no `unapply` method in object `Reset`? This is basically what the compiler would generate if `Reset` were a case class, IIRC.

Reply:
Ah, ignore me. I confused myself while jumping between the trait and the `case class`es implementing it. Hope it works fine on Scala 3 and it changes to `unapply` / introduction of `Matchable`.